source/data-modeling/persistence-configuration.txt

```diff
@@ -0,0 +1,252 @@
+  :language: ruby
+  :start-after: start read configuration
+  :end-before: end read configuration
+
+Setting Persistence Contexts Globally
++++++++++++++++++++++++++++++++++++++
+
+{+odm+} provides the ability to globally define the persistence configuration at
+the database and client level. This is helpful if you want to switch the 
+persistence context for all operations at runtime but don't want to use ``with`` 
+multiple times throughout your code.
+
+To globally define the persistence context in your program, use the
+``{+odm+}.override_client`` and ``{+odm+}.override_database`` methods. For 
+example, if your application stores information for different locales in 
+different databases, but the overall schema remains the same, you
+can use the ``{+odm+}.override_database`` function:
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+  :language: ruby
+  :start-after: start global configuration example
+  :end-before: end global configuration example
+
+In this example, all remaining operations on this thread will be performed on 
+an alternative database determined by the locale. Because the ``after_action`` 
+macro sets the override option to ``nil``, subsequent requests with no
+changes in persistence configuration will use the default configuration.
+
+Client and Collection Access
+----------------------------
+
+You can access the client or collection of a model or document instance using
+the ``mongo_client`` and ``collection`` class functions:
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+  :language: ruby
+  :start-after: start access client collection
+  :end-before: end access client collection
+
+When using these methods, you can also set runtime persistence options using
+``with``, as described in the :ref:`mongoid-set-runtime-persistence-options`
+described earlier.
+
+``mongo_client.with``
+~~~~~~~~~~~~~~~~~~~~~
+
+The following code example accesses the client used by the ``Band`` model class.
+It then uses the ``with()`` method on the client to write to the ``music``
+database, setting the ``w`` write option to ``0`` to not require write acknowledgement.
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+  :language: ruby
+  :start-after: start client with example
+  :end-before: end client with example
+
+``collection.with``
+~~~~~~~~~~~~~~~~~~~
+
+You can override the ``:read`` or ``:write`` options on a collection using the
+``with()`` method on a collection. The following example shows how to use
+the ``with()`` method on a collection to set write preferences.
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+  :language: ruby
+  :start-after: start collection with example
+  :end-before: end collection with example 
+
+API Documentation
+-----------------
+
+Check out the following API documentation for more information on the methods
+mentioned in this guide:
```

Comment:
I would love to write fun and casual docs like this, but that's not the current style 😞 
also [on/about](https://www.mongodb.com/docs/meta/style-guide/terminology/alphabetical-terms/#:~:text=Don%27t%20use%20on%20to%20mean%20about.)

```suggestion
For more information about the methods mentioned in this guide, see the following API documentation:
```